src/datasets/utils/file_utils.py

```diff
@@ -255,12 +254,16 @@ def get_authentication_headers_for_url(
         token = use_auth_token
     headers = {}
     if url.startswith(config.HF_ENDPOINT):
-        if token is False:
+        if config.HF_HUB_VERSION >= version.parse("0.20.0"):
+            return huggingface_hub.utils.build_hf_headers(
+                token=token, library_name="datasets", library_version=__version__
+            )
```

Comment:
`build_hf_headers` is available since Sept 2022 (see https://github.com/huggingface/huggingface_hub/pull/1064) so no need to check `HF_HUB_VERSION` here

Reply:
indeed, therefore I got rid of HfFolder completely :)